internal/engine/compiler/engine.go

```diff
@@ -1368,8 +1334,54 @@ func compileWasmFunction(cmp compiler, ir *wazeroir.CompilationResult) (*code, e
 			err = cmp.compileV128Narrow(o)
 		case wazeroir.OperationV128ITruncSatFromF:
 			err = cmp.compileV128ITruncSatFromF(o)
-		case wazeroir.OperationBuiltinFunctionCheckExitCode:
-			err = cmp.compileBuiltinFunctionCheckExitCode()
+		case wazeroir.OperationNullary:
+			switch op.Kind() {
```

Comment:
what I meant is like this

```suggestion
		case wazeroir.UnionOperation:
			switch op.Kind {
```

instead of define "new" implementation of old Operation interface.